torchvision/transforms/v2/_geometry.py

```diff
@@ -752,8 +752,7 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         if isinstance(output, datapoints.BoundingBox):
             # We "mark" the invalid boxes as degenreate, and they can be
             # removed by a later call to SanitizeBoundingBoxes()
-            output[params["is_within_crop_area"]] = 0
-            output = datapoints.BoundingBox.wrap_like(output, inpt)
+            output[~params["is_within_crop_area"]] = 0
```

Comment:
Three issues here:

1. We want to mark the boxes *not* within the area as degenerate
2. Wrapping with `datapoints.BoundingBox.wrap_like(output, inpt)` just copies the metadata from `output` onto `inpt`. However, this ignores that we use `F.crop` above which changes the values as well. So it should be `datapoints.BoundingBox.wrap_like(inpt, output, spatial_size=spatial_size)` where `spatial_size` needs to be extracted before.
  Plot twist: this is an inplace op, so the type is persistent and we don't need to wrap at all. So we can just remove this line completely.